src/malli/core.cljc

```diff
@@ -830,6 +905,20 @@
                                          (identical? v* v) x
                                          :else (assoc x i v*))))
                                    x parsers)))))
+          (-unparser [_]
```

Comment:
the `-unparser` code are basically 95% of copy-paste from `-parser`. we could share code in the future.

Reply:
I made a sketch https://github.com/metosin/malli/compare/master...nilern:coercer a while back about coercion (validation + transform like in Plumatic Schema) and I believe that approach could also yield `parse`, `unparse`, `encode` and `decode` as special cases... but it gets complicated.

Reply:
Having one generic solution would be great - if that emits code that is not (much) slower than the current solution. Currently, for example. `-validate´ tries to be close to optimal clojure code having no extra args or function calls. JVM inlining works well with small and simple methods.

Reply:
There shouldn't be such issues if it is the code that constructs the parsers, unparsers etc. that is shared instead of constructing one coercer function that implements everything by branching on extra arguments or something.